src/server/editorServices.ts

```diff
@@ -2069,6 +2067,27 @@ namespace ts.server {
             return this.openClientFileWithNormalizedPath(toNormalizedPath(fileName), fileContent, scriptKind, /*hasMixedContent*/ false, projectRootPath ? toNormalizedPath(projectRootPath) : undefined);
         }
 
+        /**
+         * @internal
+         * Returns undefined if the file does not exist.
+         */
+        tryOpeningProjectForFileIfNecessary(fileName: NormalizedPath): { readonly fileJustOpened: boolean, readonly project: Project | undefined } | undefined {
+            const scriptInfo = this.filenameToScriptInfo.get(fileName);
+            const getProject = (configFileName: NormalizedPath | undefined) => configFileName === undefined ? undefined : this.getOrCreateConfiguredProject(configFileName).project;
+            if (scriptInfo && scriptInfo.isScriptOpen()) {
+                return { fileJustOpened: false, project: getProject(this.getConfigFileNameForFile(scriptInfo)) };
+            }
+            else {
+                const openResult = this.tryOpenClientFileWithNormalizedPath(fileName, /*fileContent*/ undefined, /*scriptKind*/ undefined, /*hasMixedContent*/ undefined, /*projectRootPath*/ undefined, /*dontOpenIfNotExists*/ true);

```

Comment:
Why do you need to open the script info here? This might break asserts if editor opens file after this operation? (we also need to worry about projectRoot when opening a file which is not handled here)

Reply:
If I change this to:
```ts
const getResult = this.getOrCreateScriptInfoOpenedByClientForNormalizedPath(fileName, this.currentDirectory, /*fileContent*/ undefined, /*scriptKind*/ undefined, /*hasMixedContent*/ undefined, /*openIfNotExists*/ false);
return getResult && { fileJustOpened: true, project: getProject(this.getConfigFileNameForFile(getResult)) };
```
There will be an error at `Debug.assert(this.openFiles.has(info.path))` in `forEachConfigFileLocation`. Could you take a look?

Reply:
I've modified `forEachConfigFileLocation` to not have that assert in this case. If the file isn't open `projectRootPath` will be undefined, so `anySearchPathOk` will be true and it should look in all possible locations.